src/Tasks/SynTree/Config.hs

```diff
@@ -110,7 +130,21 @@ checkSynTreeConfig SynTreeConfig {..}
     | minUniqueBinOperators > fromIntegral (length [minBound .. maxBound :: BinOp]) = reject $ do
         english "The number of unique operators cannot exceed the maximum number of operators."
         german "Die Anzahl der unterschiedlichen Operatoren kann nicht die maximale Anzahl überschreiten."
-    | not allowArrowOperators && minUniqueBinOperators > 2 = reject $ do
-        english "This number of unique operators cannot be reached with allowArrowOperators = False ."
-        german "Die angegebene Anzahl der unterschiedlichen Operatoren kann mit allowArrowOperators = False nicht erreicht werden."
+    | any (< 0) operatorFrequencies = reject $ do
+        english "The operator frequencies must be non-negative."
+        german "Die Frequenzen für die Operatoren müssen nicht-negativ sein."
+    | all (== 0) operatorFrequencies = reject $ do
+        english "At least one operator must have a positive frequency."
+        german "Mindestens ein Operator muss eine positive Frequenz haben."
+    | minUniqueBinOperators > fromIntegral (length (filter (> 0) [and, or, impl, backImpl, equi])) = reject $ do
+        english "The minimum number of different operators exceeds the maximum number of available operators."
+        german "Die minimale Anzahl unterschiedlicher Operatoren überschreitet die maximale Anzahl an verfügbaren Operatoren."
```

Comment:
I guess there should also be some check(s?) here relating `maxConsecutiveNegations` and the `neg` frequency setting.

Reply:
Is there a bigger connection than "`maxConsecutiveNegations` is useless unless `neg frequency > 0`"? I don't see how they are related otherwise.